cmd/loom/gateway_reactors.go

```diff
@@ -17,21 +19,51 @@ func startGatewayReactors(
 	cfg *config.Config,
 	nodeSigner glAuth.Signer,
 ) error {
+	if err := checkQueryService(cfg.TransferGateway.DAppChainReadURI, cfg.QueryServicePollTimeout); err != nil {
+		return err
+	}
```

Comment:
You need to move this into `startGatewayFn`, and only ping the query server if the batch signing is actually enabled in the config.